### PR TITLE
test: Replace gArgs with local argsman in bench

### PR DESCRIPTION
**Pull request by maflcko**
All utilities use the same gArgs global that the node uses. This is generally fine and does not lead to issues unless a bench test is going to spin up a NodeContext via the TestingSetup. In that case the two uses of gArgs conflict and currently it needs to be cleared:

https://github.com/bitcoin/bitcoin/blob/544709763e1f45148d1926831e07ff03487673ee/src/bench/bench_bitcoin.cpp#L76

One solution would be to do nothing, because the current code works with that workaround. Another solution would be to not use the same global in all binaries.

### CODE REVIEW

File: src/util/settings.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include <string>
 #include <vector>
 
-class UniValue;
+#include <univalue.h> // For util::SettingsValue = UniValue
```
Comment (promag) [inline src/util/settings.h:12]: Why this change? Isn't forward declaration enough?

**Review by promag — COMMENTED** [review]
Code review ACK fae00a77e2589cc784650e3e60e1b99c22ca8a7b.

**Comment by practicalswift** [comment]
Concept ACK

On behalf of future generations on Bitcoin Core developers: thanks for doing these cleanups - that will make their lives easier :)

The short-term gain from each individual clean-up might seem small but the cumulative long-term gain will make a big difference.

File: src/util/settings.h (hunk 1 above)
Comment (MarcoFalke) [reply src/util/settings.h:12]: It doesn't compile without this change

Comment (promag) [reply src/util/settings.h:12]: But include just in src/bench/bench_bitcoin.cpp?

Comment (MarcoFalke) [reply src/util/settings.h:12]: Reason is that SettingsValue is a return value of one of the methods of ArgsManager. Not sure why C++ doesn't require the struct layout for return values of methods when no object with such method is initialized. Though it does require the layout when an object with such method is instantiated... I can't change the way C++ works.

Comment (MarcoFalke) [reply src/util/settings.h:12]: > But include just in src/bench/bench_bitcoin.cpp?

Yes, that works, but it feels weird to require to include implementation details of one class on the module that ends up happening to use the class.

Comment (ryanofsky) [reply src/util/settings.h:12]: re: https://github.com/bitcoin/bitcoin/pull/18662#discussion_r409486918

> Reason is that SettingsValue is a return value of one of the methods of ArgsManager. Not sure why C++ doesn't require the struct layout for return values of methods when no object with such method is initialized. Though it does require the layout when an object with such method is instantiated... I can't change the way C++ works.

Complier error messages here are pretty unhelfpul, but this happens because of the ArgsManager destructor, not the method return value. Following change should fix:

```diff
diff --git a/src/util/settings.h b/src/util/settings.h
index bbb6abe2c09..1d03639fa26 100644
--- a/src/util/settings.h
+++ b/src/util/settings.h
@@ -9,7 +9,7 @@
 #include <string>
 #include <vector>
 
-#include <univalue.h> // For util::SettingsValue = UniValue
+class UniValue;
 
 namespace util {
 
diff --git a/src/util/system.cpp b/src/util/system.cpp
index b0a538b5277..49a3ab447b9 100644
--- a/src/util/system.cpp
+++ b/src/util/system.cpp
@@ -231,6 +231,8 @@ ArgsManager::ArgsManager()
     // nothing to do
 }
 
+ArgsManager::~ArgsManager() {}
+
 const std::set<std::string> ArgsManager::GetUnsuitableSectionOnlyArgs() const
 {
     std::set<std::string> unsuitables;
diff --git a/src/util/system.h b/src/util/system.h
index 3138522b5c8..96f51e6074a 100644
--- a/src/util/system.h
+++ b/src/util/system.h
@@ -192,6 +192,7 @@ protected:
 
 public:
     ArgsManager();
+    ~ArgsManager();
 
     /**
      * Select the network in use
```

**Review by ryanofsky — APPROVED** [review]
Code review ACK fae00a77e2589cc784650e3e60e1b99c22ca8a7b

File: src/util/settings.h (hunk 1 above)
Comment (MarcoFalke) [reply src/util/settings.h]: Thanks. Pushed.

**Review by ryanofsky — APPROVED** [review]
Code review ACK faf989f93695d29099f6e152d5a2e117cd304183. Just new commit added restoring forward declaration

**Comment by promag** [comment]
ACK faf989f93695d29099f6e152d5a2e117cd304183.
